bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/discovery/PilightDeviceDiscoveryService.java

```diff
@@ -152,20 +149,18 @@ protected void stopBackgroundDiscovery() {
 
     @Override
     public void initialize() {
-        bridgeUID = thingHandler.getThing().getUID();
-        boolean discoveryEnabled = false;
         removeOlderResults(new Date().getTime(), thingHandler.getThing().getUID());
-        discoveryEnabled = thingHandler.isBackgroundDiscoveryEnabled();
         thingHandler.registerDiscoveryListener(this);
 
         super.initialize();
-        super.modified(Map.of(DiscoveryService.CONFIG_PROPERTY_BACKGROUND_DISCOVERY, discoveryEnabled));
+        super.modified(Map.of(DiscoveryService.CONFIG_PROPERTY_BACKGROUND_DISCOVERY,
+                thingHandler.isBackgroundDiscoveryEnabled()));
     }
 
     @Override
     public void dispose() {
         super.dispose();
-        removeOlderResults(getTimestampOfLastScan(), bridgeUID);
+        removeOlderResults(new Date().getTime(), thingHandler.getThing().getUID());
```

Comment:
You could remove this usage of the old date/time API:
```suggestion
        removeOlderResults(Instant.now().getEpochSecond(), thingHandler.getThing().getUID());
```

Reply:
Shouldn't this be Instant.now().toEpochMilli(), since Date().getTime()  
> Returns the number of **milliseconds** since January 1, 1970, 00:00:00 GMT represented by this Date object.

Reply:
Yes! Good catch.

Reply:
Done